### PR TITLE
apps: build versioning, add `list-buildpacks` and `upgrade-buildpack`

### PR DESCRIPTION
**Pull request by kamaln7**

### CODE REVIEW

File: commands/apps.go [hunk 1]
```diff
@@ -225,6 +225,32 @@ Only basic information is included with the text output format. For complete app
...
+		cmd,
+		RunAppUpgradeBuildpack,
+		"upgrade-buildpack <app id>",
+		"Upgrade buildpack",
+		`Upgrade a buildpack for an app`,
+		Writer,
+		displayerType(&displayers.Deployments{}),
+	)
+	AddStringFlag(upgradeBuildpack,
+		doctl.ArgBuildpack, "", "", "The ID of the buildpack to upgrade. Required.", requiredOpt())
```
Comment (kamaln7) [inline commands/apps.go:248]: Buildpack ID as a flag and not an arg to leave support for a future `upgrade-buildpack` where the ID may be optional to apply any available upgrades possibly for multiple buildpacks.

File: go.mod [hunk 2]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/containerd/continuity v0.3.0 // indirect
 	github.com/creack/pty v1.1.11
-	github.com/digitalocean/godo v1.87.0
+	github.com/digitalocean/godo v1.87.1-0.20221028144417-7dec38659ec7
```
Comment (kamaln7) [inline go.mod:10]: waiting on https://github.com/digitalocean/godo/pull/574

File: commands/apps.go [hunk 3]
```diff
@@ -225,6 +225,32 @@ Only basic information is included with the text output format. For complete app
...
+		cmd,
+		RunAppUpgradeBuildpack,
+		"upgrade-buildpack <app id>",
+		"Upgrade buildpack",
+		`Upgrade a buildpack for an app`,
+		Writer,
+		displayerType(&displayers.Deployments{}),
+	)
+	AddStringFlag(upgradeBuildpack,
+		doctl.ArgBuildpack, "", "", "The ID of the buildpack to upgrade", requiredOpt())
```
Comment (nicktate) [inline commands/apps.go:248]: nit: Might be worth calling out in help messaging they can list the existing versions with the `list-buildpacks` command.

**Review by nicktate — APPROVED** [review]

**Review by bentranter — COMMENTED** [review]
Code looks good, I'll approve and merge after the godo version bump is done 👍

File: go.mod (hunk 2 above)
Comment (andrewsomething) [reply go.mod:10]: https://github.com/digitalocean/godo/releases/tag/v1.88.0

File: commands/apps.go (hunk 3 above)
Comment (kamaln7) [reply commands/apps.go:248]: good idea

File: commands/command.go [hunk 4]
```diff
@@ -96,7 +96,7 @@ func cmdBuilderWithInit(parent *Command, cr CmdRunner, cliText, shortdesc string
 	}
 
 	if cols := c.fmtCols; cols != nil {
-		formatHelp := fmt.Sprintf("Columns for output in a comma-separated list. Possible values: `%s`",
+		formatHelp := fmt.Sprintf("Columns for output in a comma-separated list. Possible values: ```%s`.",
```
Comment (kamaln7) [inline commands/command.go:99]: It looks like Cobra uses the first text within backquotes as the format for the flag.

before:

```
      --format ID            Columns for output in a comma-separated list. Possible values: ID, `Cause`, `Progress`, `Created`, `Updated`
```

after:

```
      --format               Columns for output in a comma-separated list. Possible values: `ID`, `Cause`, `Progress`, `Created`, `Updated`.
```

**Review by nicktate — APPROVED** [review]

File: commands/command.go (hunk 4 above)
Comment (andrewsomething) [reply commands/command.go:99]: Oh, this is a good trick. Might need to do this in some other places. I've always found this Cobra behavior annoying.

**Review by andrewsomething — APPROVED** [review]
:+1:  LGTM

**Pull request merged**